assets/src/hooks/useDetour.ts

```diff
@@ -27,37 +28,76 @@ export enum DetourState {
   Finished,
 }
 
-export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
-  const [snapshot, send] = useMachine(createDetourMachine)
+const selectFinishedState = (
+  snapshot: SnapshotFrom<typeof createDetourMachine>
+) => {
+  /*
+   * There's probably a better way to do this? Tags or maybe context?
+   * Tags seem more appropriate, but weird to manage Out-Of-Bounds state via tags.
+   * Maybe this entire API call could be moved to and managed by an actor
+   * combined with parallel child states within the state machine?
+   * -- https://stately.ai/docs/promise-actors */
+  const isFinishedDrawing = snapshot.matches({
+    "Detour Drawing": { Editing: "Finished Drawing" },
+  })
+  const isSharingDetour = snapshot.matches({ "Detour Drawing": "Share Detour" })
+  const isInFinishedDetourState = isFinishedDrawing || isSharingDetour
+  return isInFinishedDetourState
+}
+
+/** This selects the detour waypoints in-between the detour start and end points  */
+const selectWaypoints = (snapshot: SnapshotFrom<typeof createDetourMachine>) =>
+  snapshot.matches({
+    "Detour Drawing": { Editing: "Finished Drawing" },
+  })
+    ? snapshot.context.waypoints.slice(1, -1)
+    : snapshot.matches({ "Detour Drawing": { Editing: "Place Waypoint" } })
+    ? snapshot.context.waypoints.slice(1)
+    : []
+
+const selectStartPoint = (snapshot: SnapshotFrom<typeof createDetourMachine>) =>
+  snapshot.context.waypoints.at(0)
 
-  const [startPoint, setStartPoint] = useState<ShapePoint | null>(null)
-  const [endPoint, setEndPoint] = useState<ShapePoint | null>(null)
-  const [waypoints, setWaypoints] = useState<ShapePoint[]>([])
+const selectEndPoint = (snapshot: SnapshotFrom<typeof createDetourMachine>) =>
+  (selectFinishedState(snapshot) && snapshot.context.waypoints.at(-1)) ||
+  undefined
+
+export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
+  const [snapshot, send, createDetourActor] = useMachine(createDetourMachine)
+  const startPoint = useSelector(createDetourActor, selectStartPoint)
```

Comment:
Curiosity/Nit: How come you changed the way that waypoints/start point/end point are stored? 

I'm partly asking because I had been thinking of something like:
- From the `Pick Start Point` state, if the event `detour.edit.place-connection-point` happens, then it initializes the `startPoint` context variable and switches to the `Place Waypoint` state.
- From the `Place Waypoint` state, if the event `detour.edit.place-connection-point` happens, then it initializes the `endPoint` context variable and switches to the `Finished Detour` final state (and then shifts out of `Editing`.

That feels a bit more ... state machine-y to me, since the states then have tighter control over when the start point/end point/etc get set.

Reply:
Thinking this through more....

We do need `startPoint`/`endPoint`/`waypoints` separately for rendering, and also we need `allPoints` (below) for directions. 

Given the choice between computing `allPoints` from `startPoint`/`waypoints`/`endPoint` versus computing `startPoint`/`waypoints`/`endPoint` from `allPoints`, I think I prefer concatenating over slicing.

But I wonder how evil it would be to just track `allPoints` separately from `sP`/`wp`/`eP`? I'm torn, because it would be nice to Just Have™ `allPoints` handed to us from the state machine, but also as soon as we start storing the same info twice, we run the risk of goofing and letting things get out of sync.